src/types/Typechecker/Suggestable.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Typechecker.Suggestable (
+                                Suggestable
+                                ,smallSuggest
+                                ,longSuggest
+                                )where
+
+-- Library dependencies
+import Text.PrettyPrint.Annotated
+import Text.Printf (printf)
+import Data.Maybe
+
+-- Module dependencies
+import AST.AST
+import AST.PrettyPrinter hiding (indent)
+import Typechecker.TypeError
+import Typechecker.Environment
+import Typechecker.Util
+import Identifiers
+import Types
+
+
+pipe = char '|'
+
+
+makeNotation :: Doc TCStyle
+makeNotation = styleLogistic (pipe $+$ equals) <+> styleDesc (text "note:")
+
+-- How to determine if to use a smallSuggest or longSuggest:
```

Comment:
I could definitely add some documentation.

What do you mean version1 or version2? the codeviewer?
There are two versions right now doing the same thing, one is ment to be removed and amended before merging.
It's made so that you could construct different codeviewers and choose based on the error
(an example is if you have the wrong arguments for a function, the code viewer could display the actual function declaration as well as where the programmer wrote wrong. I think the declaration s range also includes the entire body which is why I never made such a viewer)

The small and long suggest will always print, which is why default is empty (nothing will print).
A desire that I included as future work was to refactor the suggestions to be returned as one function, meaning that you could evaluate the code, and depending on how much information you want to relay back you could then choose if it would be displayed as a short or long suggest (or both). Right now you would need to compute this for both functions individually.